src/hotspot/share/opto/superword.cpp

```diff
@@ -1630,36 +1621,88 @@ uint SuperWord::max_implemented_size(const Node_List* pack) {
   }
 }
 
-// Java API for Long.bitCount/numberOfLeadingZeros/numberOfTrailingZeros
-// returns int type, but Vector API for them returns long type. To unify
-// the implementation in backend, superword splits the vector implementation
-// for Java API into an execution node with long type plus another node
-// converting long to int.
-bool SuperWord::requires_long_to_int_conversion(int opc) {
-  switch(opc) {
-    case Op_PopCountL:
-    case Op_CountLeadingZerosL:
-    case Op_CountTrailingZerosL:
-      return true;
-    default:
-      return false;
+// If the j-th input for all nodes in the pack is the same unique input: return it, else nullptr.
+Node* PackSet::isa_unique_input_or_null(const Node_List* pack, int j) const {
+  Node* p0 = pack->at(0);
+  Node* unique = p0->in(j);
+  for (uint i = 1; i < pack->size(); i++) {
+    if (pack->at(i)->in(j) != unique) {
+      return nullptr; // not unique
+    }
   }
+  return unique;
 }
 
-//------------------------------same_inputs--------------------------
-// For pack p, are all idx operands the same?
-bool SuperWord::same_inputs(const Node_List* p, int idx) const {
-  Node* p0 = p->at(0);
-  uint vlen = p->size();
-  Node* p0_def = p0->in(idx);
-  for (uint i = 1; i < vlen; i++) {
-    Node* pi = p->at(i);
-    Node* pi_def = pi->in(idx);
-    if (p0_def != pi_def) {
-      return false;
+VTransformBoolTest PackSet::get_bool_test(const Node_List* bool_pack) const {
+  BoolNode* bol0 = bool_pack->at(0)->as_Bool();
+  BoolTest::mask mask = bol0->_test._test;
+  bool is_negated = false;
+  assert(mask == BoolTest::eq ||
+         mask == BoolTest::ne ||
+         mask == BoolTest::ge ||
+         mask == BoolTest::gt ||
+         mask == BoolTest::lt ||
+         mask == BoolTest::le,
+         "Bool should be one of: eq,ne,ge,ge,lt,le");
```

Comment:
Add spaces and fix typo:
```suggestion
         "Bool should be one of: eq, ne, ge, gt, lt, le");
```